### PR TITLE
feat: [CDS-41717]: Add support to PowerShell use proxy env variable

### PR DESCRIPTION
**Pull request by Fernando-Dourado**
## Describe your changes
- Add support to use the `HTTP_PROXY` environment variable inside a PowerShell to download from the artifactory service.

- The change is behind the environment variable `POWERSHELL_USE_ENV_PROXY` used inside the delegate. That way we reduce the changes and any back compatibility side effects.

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/38463)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/38463)

File: 930-delegate-tasks/src/main/java/software/wings/beans/delegation/ShellScriptParameters.java [hunk 1]
```diff
@@ -105,6 +105,7 @@ public class ShellScriptParameters implements TaskParameters, ActivityAccess, Ex
   private Boolean includeInfraSelectors;
   private boolean enableJSchLogs;
   private Integer sshTimeOut;
+  boolean powerShellUseEnvironmentProxy;
```
Comment (abhinav-harness) [inline 930-delegate-tasks/src/main/java/software/wings/beans/delegation/ShellScriptParameters.java:108]: probably Boolean for bg compatible?

Comment (Fernando-Dourado) [reply 930-delegate-tasks/src/main/java/software/wings/beans/delegation/ShellScriptParameters.java:108]: It's really required? Let me confirm with someone from delegate team.

Or do you recommend the change?

Comment (abhinav-harness) [reply 930-delegate-tasks/src/main/java/software/wings/beans/delegation/ShellScriptParameters.java:108]: yeah we should do Boolean to be safe

**Review by abhinav-harness — DISMISSED** [review]
> Reviewed 2 of 8 files at r1, 6 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38463)* status: :shipit: complete! ✓

**Comment by abhinav-harness** [comment]
trigger codebasehash

**Review by arvind-choudhary-h — APPROVED** [review]
> Reviewed 2 of 8 files at r1, 6 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38463)* status: :shipit: complete! ✓

**Review by shubham-in10se — CHANGES_REQUESTED** [review]
> Reviewed 2 of 8 files at r1, 6 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38463)* status: :shipit: complete! ✓

___
*[`930-delegate-tasks/src/main/java/software/wings/beans/command/CommandExecutionContext.java` line 110 at r2](https://reviewable.io/reviews/harness/harness-core/38463#-NE6OeIV0qHMFUrmx7Zs:-NE6OeIV0qHMFUrmx7Zt:bspvd0a) ([raw file](https://github.com/harness/harness-core/blob/eac0a71129990f809217ba4665428c6b5169a206/930-delegate-tasks/src/main/java/software/wings/beans/command/CommandExecutionContext.java#L110)):*
> ```Java
>   private boolean useWinRMKerberosUniqueCacheFile;
>   private List<String> delegateSelectors;
>   private Boolean powerShellUseEnvironmentProxy;
> ```

using `Boolean` in place of native datatype `boolean` means you've have to apply null checks as well for this variable everywhere. 
If we are good with `powerShellUseEnvironmentProxy` having a default value of `false`, let's prefer using the native datatype here.



_Code quote:_
```Java
Boolean 
```
___
*[`930-delegate-tasks/src/main/java/software/wings/beans/delegation/ShellScriptParameters.java` line 108 at r2](https://reviewable.io/reviews/harness/harness-core/38463#-NE6PFsbBH5YdNYKzlDS:-NE6PFsbBH5YdNYKzlDT:b2a9uj2) ([raw file](https://github.com/harness/harness-core/blob/eac0a71129990f809217ba4665428c6b5169a206/930-delegate-tasks/src/main/java/software/wings/beans/delegation/ShellScriptParameters.java#L108)):*
> ```Java
>   private boolean enableJSchLogs;
>   private Integer sshTimeOut;
>   Boolean powerShellUseEnvironmentProxy;
> ```

should be private



_Code quote:_
```Java
Boolean powerShellUseEnvironmentProxy;
```

**Review by shubham-in10se — APPROVED** [review]
> Reviewed 8 of 8 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38463)* status: :shipit: complete! ✓

**Review by puthrayaharness — APPROVED** [review]

**Comment by puthrayaharness** [comment]
trigger codebasehash

**Review by abhinav-harness — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38463)* status: :shipit: complete! ✓

**Review by abhinav-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38463)* status: :shipit: complete! ✓

___
*[`930-delegate-tasks/src/main/java/software/wings/beans/command/DownloadArtifactCommandUnit.java` line 560 at r3](https://reviewable.io/reviews/harness/harness-core/38463#-NECCuNB1yHn7n7R3Aj1:-NECCuNB1yHn7n7R3Aj2:bpq10nz) ([raw file](https://github.com/harness/harness-core/blob/ba89f01123ce26bb594e0c252191c14b619aa189/930-delegate-tasks/src/main/java/software/wings/beans/command/DownloadArtifactCommandUnit.java#L560)):*
> ```Java
>     final String value = SystemWrapper.getenv(POWERSHELL_USE_ENV_PROXY);
>     return Boolean.parseBoolean(value);
>   }
> ```

generally i like string comparision with matching case as customer can write - "true","True" or even "TRUE"

**Comment by Fernando-Dourado** [comment]
> > _[Reviewable](https://reviewable.io/reviews/harness/harness-core/38463)_ status: :shipit: complete! ✓
> 
> _[`930-delegate-tasks/src/main/java/software/wings/beans/command/DownloadArtifactCommandUnit.java` line 560 at r3](https://reviewable.io/reviews/harness/harness-core/38463#-NECCuNB1yHn7n7R3Aj1:-NECCuNB1yHn7n7R3Aj2:bpq10nz) ([raw file](https://github.com/harness/harness-core/blob/ba89f01123ce26bb594e0c252191c14b619aa189/930-delegate-tasks/src/main/java/software/wings/beans/command/DownloadArtifactCommandUnit.java#L560)):_
> 
> > ```java
> >     final String value = SystemWrapper.getenv(POWERSHELL_USE_ENV_PROXY);
> >     return Boolean.parseBoolean(value);
> >   }
> > ```
> 
> generally i like string comparision with matching case as customer can write - "true","True" or even "TRUE"

The `parseBoolean` already does an equalsIgnoreCase evaluation.

```
public static boolean parseBoolean(String s) {
  return "true".equalsIgnoreCase(s);
}
```

**Comment by abhinav-harness** [comment]
*[`930-delegate-tasks/src/main/java/software/wings/beans/command/DownloadArtifactCommandUnit.java` line 560 at r3](https://reviewable.io/reviews/harness/harness-core/38463#-NECCuNB1yHn7n7R3Aj1:-NECSFwk43kLU4nii37H:bc2f03o) ([raw file](https://github.com/harness/harness-core/blob/ba89f01123ce26bb594e0c252191c14b619aa189/930-delegate-tasks/src/main/java/software/wings/beans/command/DownloadArtifactCommandUnit.java#L560)):*
<details><summary><i>Previously, Fernando-Dourado (Fernando Dourado) wrote…</i></summary><blockquote>

The `parseBoolean` already does an equalsIgnoreCase evaluation.

```
public static boolean parseBoolean(String s) {
  return "true".equalsIgnoreCase(s);
}
```

</blockquote></details>

:cool:

**Pull request merged**